src/audio/selector/selector.c

```diff
@@ -83,6 +91,140 @@ static struct comp_dev *selector_new(const struct comp_driver *drv,
 	return dev;
 }
 
+static void set_selector_params(struct comp_dev *dev,
+				struct sof_ipc_stream_params *params)
+{
+}
+#else
+static void build_config(struct comp_data *cd)
+{
+	enum sof_ipc_frame valid_format;
+
+	cd->source_format = cd->md.base_cfg.audio_fmt.depth;
+	audio_stream_fmt_conversion(cd->md.base_cfg.audio_fmt.depth,
+				    cd->md.base_cfg.audio_fmt.valid_bit_depth,
+				    &cd->source_format,
+				    &valid_format,
+				    cd->md.base_cfg.audio_fmt.s_type);
+
+	audio_stream_fmt_conversion(cd->md.output_format.depth,
+				    cd->md.output_format.valid_bit_depth,
+				    &cd->sink_format,
+				    &valid_format,
+				    cd->md.output_format.s_type);
```

Comment:
As line 104 will set cd->source_format, is line 103 necessary? What's more, cd->source_format/cd->sink_format will be set again in prepare(), is it necessary to set source_format and sink_format here?

Reply:
cd->source_format is in IPC4 mode in 103 and it will be converted to IPC3 mode in 104. We set sink & source source_format and sink format based on cd format. it is not duplicated.